frontend/src/scenes/web-analytics/pageReportsLogic.ts

```diff
@@ -0,0 +1,343 @@
+                                        FROM events
+                                        WHERE event = '$pageview'
+                                        AND cutQueryStringAndFragment(properties.$current_url) like '%${hogql.identifier(
+                                            searchTerm
+                                        )}%'
+                                        GROUP BY url
+                                        ORDER BY count DESC
+                                        LIMIT 100`
+                                    : hogql`SELECT DISTINCT properties.$current_url AS url, count() as count
+                                        FROM events
+                                        WHERE event = '$pageview'
+                                        AND properties.$current_url like '%${hogql.identifier(searchTerm)}%'
+                                        GROUP BY url
+                                        ORDER BY count DESC
+                                        LIMIT 100`,
+                            }
+                        } else {
+                            query = {
+                                kind: NodeKind.HogQLQuery,
+                                query: values.stripQueryParams
+                                    ? hogql`SELECT DISTINCT cutQueryStringAndFragment(properties.$current_url) AS url, count() as count
+                                        FROM events
+                                        WHERE event = '$pageview'
+                                        GROUP BY url
+                                        ORDER BY count DESC
+                                        LIMIT 100`
+                                    : hogql`SELECT DISTINCT properties.$current_url AS url, count() as count
+                                        FROM events
+                                        WHERE event = '$pageview'
+                                        GROUP BY url
+                                        ORDER BY count DESC
+                                        LIMIT 100`,
+                            }
+                        }
+
+                        const response = await api.query(query)
+                        const res = response as { results: [string, number][] }
+                        const results = res.results?.map((x) => ({ url: x[0], count: x[1] })) as PageURL[]
+
+                        return results
+                    } catch (error) {
+                        console.error('Error loading pages:', error)
+                        return []
+                    }
+                },
+            },
+        ],
+    }),
+
+    selectors: {
+        hasPageUrl: [(selectors) => [selectors.pageUrl], (pageUrl: string | null) => !!pageUrl],
+        isLoading: [
+            (selectors) => [selectors.pagesLoading, selectors.isInitialLoad],
+            (pagesLoading: boolean, isInitialLoad: boolean) => pagesLoading || isInitialLoad,
+        ],
+        queries: [
```

Comment:
Hmm I wonder if we should refactor this.

Instead of
* webAnalyticsLogic has a selector that creates all the queries and puts them in an array for the UI component to consume
* pageReportsLogic has a selector that takes the queries out of the array to put on an object for page reports UI consume

We could do
* webAnalyticsLogic has a selector that creates all the queries an puts them on an object
* webAnalyticsLogic has a selector that takes them from the object and puts in an array for the UI component to consume
* pageReportsLogic has a selector that takes the queries out of the *object* to put on an object for page reports to consume

That way your queries wouldn't need to be optional in `SimpleTile`

Reply:
I agree with that path, but this PR got ~2.5 times bigger after doing it because of the many tileId + tabId combinations required. 

In the end, I did convert the logic here to use `<Tiles />` as well, so I suggest putting that refactor in another PR.

WDYT?